### PR TITLE
RunTests-Sync Lock fix Port and Unit Tests

### PR DESCRIPTION
**Pull request by saikrishnav**

### CODE REVIEW

File: src/vstest.console/TestPlatformHelpers/TestRequestManager.cs [hunk 1]
```diff
@@ -220,50 +222,54 @@ public void AbortTestRun()
...
         private bool RunTests(TestRunCriteria testRunCriteria, ITestRunEventsRegistrar testRunEventsRegistrar)
         {
-            bool success = true;
-            using (this.currentTestRunRequest = this.testPlatform.CreateTestRunRequest(testRunCriteria))
+            lock (syncobject)
             {
-                this.runRequestCreatedEventHandle.Set();
-                try
+                bool success = true;
+                using (ITestRunRequest testRunRequest = this.testPlatform.CreateTestRunRequest(testRunCriteria))
```
Comment (codito) [inline src/vstest.console/TestPlatformHelpers/TestRequestManager.cs]: Can we please add a comment for this change?

File: test/vstest.console.UnitTests/TestPlatformHelpers/TestRequestManagerTests.cs [hunk 2]
```diff
@@ -0,0 +1,333 @@
+﻿using Microsoft.VisualStudio.TestPlatform.Client.RequestHelper;
```
Comment (codito) [inline test/vstest.console.UnitTests/TestPlatformHelpers/TestRequestManagerTests.cs]: Please add copyright header. Move usings inside namespace.

**Review by codito — APPROVED** [review]
Looks good with few comments.

File: src/vstest.console/TestPlatformHelpers/TestRequestManager.cs (hunk 1 above)
Comment (saikrishnav) [inline src/vstest.console/TestPlatformHelpers/TestRequestManager.cs]: added.

**Pull request merged**